### PR TITLE
fix Azure receiver not azure china

### PR DESCRIPTION
**Pull request by mo-silent**
**Description:**

Add azure china to complete AzureMonitor Receiver

**Link to tracking Issue:** 
#34315

### CODE REVIEW

File: .chloggen/mo-silent_fix-34315.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: 'bug_fix'
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: azuremonitorreceiver
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Add azure china to complete AzureMonitor Receiver
```
Comment (crobert-1) [inline .chloggen/mo-silent_fix-34315.yaml:10]:
```suggestion
note: Add Azure China as a `cloud` option.
```

**Review by crobert-1 — COMMENTED** [review]
Hello @mo-silent, can you please update the README as well to show this is a valid option for the `cloud` configuration option?

It would be good to add some kind of simple test for this as well.

**Comment by mo-silent** [comment]
Hi @crobert-1 
Sure. I've updated README to add a valid option for cloud configuration options.

**Review by crobert-1 — COMMENTED** [review]
Would you be able to add a simple test for this? (Sorry if you were still working on it, just commenting to make sure we didn't miss it.)

**Comment by mo-silent** [comment]
Hi @crobert-1 
I'm not adding a new function. I just supplemented the Azure Cloud type by adding Azure China Cloud.  So I don't think additional test code is needed, the existing test code uses defaultCloud, which is azureCloud.  Again, there is no separate test case written for azureGovernmentCloud in the original test code.

**Comment by crobert-1** [comment]
> Hi @crobert-1 I'm not adding a new function. I just supplemented the Azure Cloud type by adding `Azure China Cloud`. So I don't think additional test code is needed, the existing test code uses `defaultCloud`, which is `azureCloud`. Again, there is no separate test case written for `azureGovernmentCloud` in the original test code.

I was thinking it would be good to have a test where the configuration is set to the new configuration option `AzureChinaCloud`, and then show that the `*arm.ClientOptions` object returned by `getArmClientOptions` is properly set to use the `AzureChinaCloud`. Let me know if that isn't possible, I had assumed it was a relatively simple test to add, but I might be missing something.

**Comment by mo-silent** [comment]
Hi @crobert-1 
I see what you mean.  Have been added, called `TestAzureScraperClientOptions` test functions, and contains `AzureChinaCloud`, `AzureCloud`, `AzureGovernmentCloud`, the three cloud configuration options for the test cases.

**Comment by mo-silent** [comment]
Hi @crobert-1 
Well. `make gofmt` has been executed.

**Review by crobert-1 — APPROVED** [review]
Thanks for resolving @mo-silent!

**Review by codeboten — APPROVED** [review]
Thanks @mo-silent!

File: receiver/azuremonitorreceiver/config.go [hunk 2]
```diff
@@ -16,6 +16,7 @@ import (
 const (
 	azureCloud           = "AzureCloud"
 	azureGovernmentCloud = "AzureUSGovernment"
+	azureChinaCloud      = "AzureChinaCloud"
```
Comment (codeboten) [inline receiver/azuremonitorreceiver/config.go:19]: Would be good to add support for "AzureGermanCloud" as well, i'll open a follow up issue

**Pull request merged**